### ISSUE TITLE
Building with python2.7 - hashlib and other module errors

### ISSUE THREAD

**Issue by biopsin** [void-linux/void-packages#51308]
**First of all Im packging a non-repo application so I dont expect any support**, I eventually tried posting my issue on void.reddit but Im no longer able to login as of several months and by now I dont care anymore.

Yesterday I tried to package an update to UXP/palemoon and it failed with : [full build log](https://paste.quest/?fad3b3bdfa54805c#9QBjiHA7d8TBrbKHsuDhWQXS6KBiyE2GRGikp8924X88) 
bit supprised as previous release packaged just fine a month ago, so to isolate the issue I tried rebuilding the previous version again today and it fails too with exact same error. So Ive been looking thru last months commits and changes but I cant seem to se anything related to python2.7, setuptools, openssl that would break the hashlib for instance.
Futher on I tested installing python-setuptools python2 on host and same hashlib error and while installing noticed some exeptions too
```
python-setuptools-44.0.0_2: installed successfully.
python-2.7.18.9_1: configuring ...
Byte-compiling python code in usr/lib/python2.7...
Compiling ./usr/lib/python2.7/site-packages/cffi/_shimmed_dist_utils.py ...
  File "./usr/lib/python2.7/site-packages/cffi/_shimmed_dist_utils.py", line 16
    raise Exception("This CFFI feature requires setuptools on Python >= 3.12. The setuptools
module is missing or non-functional.") from ex

                                          ^
SyntaxError: invalid syntax
```

```
python-cffi-1.16.0_1: configuring ...
Byte-compiling python2.7 code for module cffi...
Compiling usr/lib/python2.7/site-packages/cffi/_shimmed_dist_utils.py ...
  File "usr/lib/python2.7/site-packages/cffi/_shimmed_dist_utils.py", line 16
    raise Exception("This CFFI feature requires setuptools on Python >= 3.12. The setuptools
module is missing or non-functional.") from ex

                                          ^
SyntaxError: invalid syntax

```
Any idea to what is happening here?

**Comment by classabbyamp** [comment]
python-cffi 1.16.0 doesn't support python 2.7:

> Drop support for end-of-life Python versions (2.7, 3.6, 3.7).

https://cffi.readthedocs.io/en/stable/whatsnew.html#v1-16-0

---

btw, all python 2 packages will be dropped within a month

**Comment by ahesford** [comment]
Should be fixed in dae26c0b7d8f2fa5a1069d26ecbca8453aa999e3.

**Comment by biopsin** [comment]
@classabbyamp @ahesford 
Thank you both for the reply, eventually I traced the hashlib breakage to latest python-2.7.18.9, reverting back 2.7.18.7 fixes the issue, however I dont know why yet..
